cpucounters.cpp

```diff
@@ -1124,7 +1124,7 @@ bool PCM::discoverSystemTopology()
     auto populateEntry = [&smtMaskWidth, &coreMaskWidth, &l2CacheMaskShift](TopologyEntry & entry, const int apic_id)
     {
         entry.thread_id = smtMaskWidth ? extract_bits_ui(apic_id, 0, smtMaskWidth - 1) : 0;
-        entry.core_id = extract_bits_ui(apic_id, smtMaskWidth, smtMaskWidth + coreMaskWidth - 1);
+        entry.core_id = smtMaskWidth + coreMaskWidth ? extract_bits_ui(apic_id, smtMaskWidth, smtMaskWidth + coreMaskWidth - 1) : 0;
```

Comment:
thanks, could you put brackets around (smtMaskWidth + coreMaskWidth)?

Reply:
sure, updated.